resource_elasticsearch_odfe_policy.go

```diff
@@ -0,0 +1,227 @@
+		Delete: resourceElasticsearchOdfePolicyDelete,
+		Schema: map[string]*schema.Schema{
+			"policy_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"body": &schema.Schema{
+				Type:             schema.TypeString,
+				Required:         true,
+				DiffSuppressFunc: diffSuppressPolicy,
+				StateFunc: func(v interface{}) string {
+					json, _ := structure.NormalizeJsonString(v)
+					return json
+				},
+			},
+			"primary_term": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Computed: true,
+			},
+			"seq_no": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				Computed: true,
+			},
+		},
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+	}
+}
+
+func resourceElasticsearchOdfePolicyCreate(d *schema.ResourceData, m interface{}) error {
+	if _, err := resourceElasticsearchPutOdfePolicy(d, m); err != nil {
+		return err
+	}
+
+	return resourceElasticsearchOdfePolicyRead(d, m)
+}
+
+func resourceElasticsearchOdfePolicyRead(d *schema.ResourceData, m interface{}) error {
+	policyID := d.Get("policy_id").(string)
+	policyResponse, err := resourceElasticsearchGetOdfePolicy(policyID, m)
+
+	if err != nil {
+		if elastic7.IsNotFound(err) {
+			log.Printf("[WARN] OdfePolicy (%s) not found, removing from state", policyID)
+			d.SetId("")
+			return nil
+		}
+	}
+
+	bodyString, err := json.Marshal(policyResponse.Policy)
+	// Need encapsulation as the reponse from the GET is different than the one in the PUT
+	bodyStringNormalized, _ := structure.NormalizeJsonString(fmt.Sprintf("{\"policy\": %+s}", string(bodyString)))
+
+	if err != nil {
+		return err
+	}
+	d.SetId(policyResponse.PolicyID)
+	d.Set("body", bodyStringNormalized)
+	d.Set("primary_term", policyResponse.PrimaryTerm)
+	d.Set("seq_no", policyResponse.SeqNo)
+
+	return nil
+}
+
+func resourceElasticsearchOdfePolicyDelete(d *schema.ResourceData, m interface{}) error {
+	var err error
+
+	path, err := uritemplates.Expand("/_opendistro/_ism/policies/{policy_id}", map[string]string{
+		"policy_id": d.Id(),
+	})
+	if err != nil {
+		return fmt.Errorf("error building URL path for policy: %+v", err)
+	}
+
+	switch m.(type) {
+	case *elastic7.Client:
+		client := m.(*elastic7.Client)
+		_, err = client.PerformRequest(context.TODO(), elastic7.PerformRequestOptions{
+			Method: "DELETE",
+			Path:   path,
+		})
+
+		if err != nil {
+			return fmt.Errorf("error deleting policy: %+v : %+v", path, err)
+		}
+	default:
+		err = errors.New("policy resource not implemented prior to Elastic v7")
+	}
+
+	return err
+}
+
+func resourceElasticsearchGetOdfePolicy(policyID string, m interface{}) (GetPolicyResponse, error) {
+	var err error
+	response := new(GetPolicyResponse)
+
+	path, err := uritemplates.Expand("/_opendistro/_ism/policies/{policy_id}", map[string]string{
+		"policy_id": policyID,
+	})
+
+	if err != nil {
+		return *response, fmt.Errorf("error building URL path for policy: %+v", err)
```

Comment:
Maybe return nil instead of *response ?

Reply:
nope the function return a `GetPolicyResponse`